pkg/client/kubernetes/pods.go

```diff
@@ -103,57 +101,67 @@ func GetPodLogs(ctx context.Context, podInterface corev1client.PodInterface, nam
 	return io.ReadAll(stream)
 }
 
-// CheckForwardPodPort tries to forward the <remote> port of the pod with name <name> in namespace <namespace> to
-// the <local> port. If <local> equals zero, a free port will be chosen randomly.
-// It returns true if the port forward connection has been established successfully or false otherwise.
-func (c *clientSet) CheckForwardPodPort(namespace, name string, local, remote int) error {
-	fw, stopChan, err := c.setupForwardPodPort(namespace, name, local, remote)
-	if err != nil {
-		return fmt.Errorf("could not setup pod port forwarding: %w", err)
-	}
-
+// CheckForwardPodPort tries to open a portForward connection with the passed PortForwarder.
+// It returns nil if the port forward connection has been established successfully or an error otherwise.
+func CheckForwardPodPort(fw PortForwarder) error {
 	errChan := make(chan error)
 	go func() {
 		errChan <- fw.ForwardPorts()
 	}()
-	defer close(stopChan)
 
 	select {
-	case err = <-errChan:
+	case err := <-errChan:
 		return fmt.Errorf("error forwarding ports: %w", err)
-	case <-fw.Ready:
+	case <-fw.Ready():
 		return nil
-	case <-time.After(time.Second * 5):
-		return errors.New("port forward connection could not be established within five seconds")
 	}
 }
 
-func (c *clientSet) setupForwardPodPort(namespace, name string, local, remote int) (*portforward.PortForwarder, chan struct{}, error) {
+type PortForwarder interface {
+	ForwardPorts() error
+	Ready() chan struct{}
+}
+
+// SetupPortForwarder sets up a PortForwarder which forwards the <remote> port of the pod with name <name> in namespace <namespace>
+// to the <local> port. If <local> equals zero, a free port will be chosen randomly.
```

Comment:
Maybe add:
```
// When calling ForwardPorts on the returned PortForwarder, it will run until the given context is cancelled.
// Hence, the given context should carry a timeout and should be cancelled once the forwarding is no longer needed.
```